### DISCUSSION TITLE
Integration with another tools

### DISCUSSION THREAD

**Discussion by mSALDANHAf** [DefectDojo/django-DefectDojo#6940] [General]
Good Afternoon.
We, at the Telecommunications company that I belong, are introducing DefectDojo as a tool for vulnerability analysts.
We are facing 2 issues.
The first is the need to introduce pre-processed findings via API, that we are doing that via API and using the Generics Findings Import.
The problem is that we lost the information about the vulnerability scanners that found that particular finding, that can be multiple after the vulnerability finding being correlated and normalized. So, the analysts asked us to add a column that presents the vulnerability scanners where that particular finding was found and to show if it's a internal finding or exposed to Internet Finding. We solved that by changing the model of the finding and the HTML. Was this the proper way? How can we add custom fields? How can we not mess up with future conflicting updates? We are thinking using rebase or if possible merging with the official repository.
The second problem is the need to introduce different trouble ticketing platforms. We see that the integration with Jira is a little coupled and is not being generalized. What should be the best way to abstract the integration with troubleticketing platforms? How would DefectDojo evolve with integrations?
Thanks in advance.

**Comment by damiencarol** [comment]
Two question in one ticket :) 
for the second part: use metadata. it's available through the API, you will be able to add custom attribute to the Finding

**Reply by mSALDANHAf** [reply]
Ok, thank you, that seems to be the logical way. But the problem with filtering and presenting it in the list mantains. Should we make the custom attributes also tags and filter by it?
